api4/user.go

```diff
@@ -1852,6 +1852,11 @@ func login(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.App.AttachSessionCookies(c.AppContext, w, r)
 	}
 
+	// For context see: https://mattermost.atlassian.net/browse/MM-39583
+	if c.App.Srv().License() != nil && *c.App.Srv().License().Features.Cloud {
+		c.App.AttachCWSIntermediaryLoginCookie(c.AppContext, w, r)
```

Comment:
Would suggest we name this based on the actual cookie content (instead of our usage in CWS), maybe `CloudSessionCookie`

```suggestion
		c.App.AttachCloudSessionCookie(c.AppContext, w, r)
```